dev/validate_bom__deploy.py

```diff
@@ -404,145 +400,6 @@ def add_post_deploy_statements(self, script):
     """Add any statements following "hal deploy apply"."""
     pass
 
-
-class KubernetesValidateBomDeployer(BaseValidateBomDeployer):
-  """Concrete deployer used to deploy Hal onto Google Cloud Platform.
-
-  This class is not intended to be constructed directly. Instead see the
-  free function make_deployer() in this module.
-  """
-  def __init__(self, options, metrics, **kwargs):
-    super(KubernetesValidateBomDeployer, self).__init__(
-        options, metrics, **kwargs)
-
-  @classmethod
-  def init_platform_argument_parser(cls, parser, defaults):
-    """Adds custom configuration parameters to argument parser.
-
-    This is a helper function for the free function init_argument_parser().
-    """
-    add_parser_argument(
-        parser, 'deploy_k8s_namespace', defaults, 'spinnaker',
-        help='Namespace for the account Spinnaker is deployed into.')
-
-  @classmethod
-  def validate_options_helper(cls, options):
-    """Adds custom configuration parameters to argument parser.
-
-    This is a helper function for make_deployer().
-    """
-    if options.deploy_distributed_platform != 'kubernetes':
-      return
-
-    if not options.k8s_account_name:
-      raise_and_log_error(
-          ConfigError('--deploy_distributed_platform="kubernetes" requires'
-                      ' a --k8s_account_name be configured.'))
-
-    if hasattr(options, "injected_deploy_spinnaker_account"):
-      raise_and_log_error(
-          UnexpectedError('deploy_spinnaker_account was already set to "{0}"'
-                          .format(options.injected_deploy_spinnaker_account)))
-    options.injected_deploy_spinnaker_account = options.k8s_account_name
-
-  def __get_pod_name(self, k8s_namespace, service):
-    """Determine the pod name for the deployed service."""
-    options = self.options
-    flags = ' --namespace {namespace} --logtostderr=false'.format(
-        namespace=k8s_namespace)
-    label_selector = '-l app.kubernetes.io/name={service}'.format(
```

Comment:
Realized that I added this label selection logic to the V1 deployer, so I guess we never needed it and the race condition we fixed in [this commit](https://github.com/spinnaker/buildtool/commit/c37f003238b8226469b018af7b930b410b566159) was what was preventing it from working all along?!